test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -101,20 +106,343 @@ def test_suspended_job_ded_time_calendared(self):
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid3)
+
+        logmsg = ";Job would conflict with reservation or top job"
+        self.scheduler.log_match(jid2 + logmsg)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid3, extend='x',
+                           offset=15, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid2)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid1, extend='x',
+                           offset=30, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+    def test_preempted_job_server_soft_limits(self):
+        """
+        This test confirms that a preempted job remains suspended if it has
+        violated server soft limits
+        """
+        a = {ATTR_rescavail + '.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        a = {'max_run_res_soft.ncpus': '[u:pbsuser1=2]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        p = '"express_queue, normal_jobs, server_softlimits"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        j1 = Job(TEST_USER1)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=4',
+                           ATTR_l + '.walltime': 20})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 8})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 10})
+        jid3 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid2, extend='x',
+                           offset=10, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid3)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid3, extend='x',
+                           offset=15, interval=2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        p = '"express_queue, normal_jobs"'
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        self.server.manager(MGR_CMD_UNSET, SERVER, 'max_run_res_soft.ncpus')
```

Comment:
These should be done by revert_to_defaults().  If they aren't, please fix them in the framework.  The scheduler unsets everything, so that should revert preempt_prio.  Does max_run_res_soft not get reverted? (same as other places in the test suite)